### ISSUE TITLE
Teleport player to me issues

### ISSUE THREAD

**Issue by 2000jago** [noxsicarius/Epoch-Admin-Tools#294]
Seems to be a problem teleporting player to me (again).  (Testbranch).  
I did not notice until now but may have been occurring since inclusion of return player to last position option.

Client rpt says nothing, but server says:

16:49:23 SetFace error: class CfgFaces.GravesHead_EP1.Face01 not found

16:49:23 Error: Error during SetFace - class CfgFaces.GravesHead_EP1.Face01 not found

16:50:13 "TELEPORT REVERT for player UID#76561198103448555 from [11162.4,3102.78,0.00117135] to [6519.05,14141.4,-0.0583191], 11980 meters, now at [11162.4,3102.78,0.00117135]"

16:50:13 "TELEPORT REVERT for player UID#76561198103448555 from [11162.4,3102.78,0.00117135] to [6519.05,14141.4,0.00137329], 11980 meters, now at [11162.4,3102.78,0.00117135]"

16:51:13 "TIME SYNC: Local Time set to [2013,8,3,13,51]"

16:51:18 Unit is not in cargo

16:51:28 "TELEPORT REVERT for player UID#76561198103448555 from [11162.4,3102.78,0.00108719] to [6521.54,14139.8,-0.078125], 11978 meters, now at [11162.4,3102.78,0.00108719]"

16:51:29 "TELEPORT REVERT for player UID#76561198103448555 from [11162.4,3102.78,0.00108719] to [6521.54,14139.8,-0.078125], 11978 meters, now at [11162.4,3102.78,0.00108719]"

16:51:30 "TELEPORT REVERT for player UID#76561198103448555 from [11162.4,3102.78,0.00108719] to [6521.54,14139.8,-0.078125], 11978 meters, now at [11162.4,3102.78,0.00108719]"

16:51:30 "TELEPORT REVERT for player UID#76561198103448555 from [11162.4,3102.78,0.00108719] to [6522.38,14139.6,-0.0521545], 11977 meters, now at [11162.4,3102.78,0.00108719]"

16:51:30 "TELEPORT REVERT for player UID#76561198103448555 from [11162.4,3102.78,0.00108719] to [6522.32,14139.5,0.00137329], 11977 meters, now at [11162.4,3102.78,0.00108719]"

16:52:20 "TELEPORT REVERT for player UID#76561198103448555 from [11162.4,3102.78,0.00202835] to [6520.96,14138.2,-0.0521851], 11976 meters, now at [11162.4,3102.78,0.00202835]"

16:52:20 "TELEPORT REVERT for player UID#76561198103448555 from [11162.4,3102.78,0.00202835] to [6520.34,14137.9,-0.065979], 11976 meters, now at [11162.4,3102.78,0.00202835]"

**Comment by noxsicarius** [comment]
Did this happen after using the return player or not? The revert part is from the antihack and I can fix that up. How much of a spam does it cause? Is it a few lines, a few hundred, or constant?

**Comment by 2000jago** [comment]
Update:  After a fresh restart, if I teleport the player, it works the FIRST time only.

If I teleport him again, he appears at my location for a second then he's back at the ORIGINAL location.  ie where he was at login time before the first successful teleport. NOT where he arrived at after the first successful teleport.

The server log says:
 0:53:11 SetFace error: class CfgFaces.GravesHead_EP1.Face01 not found
 0:53:11 Error: Error during SetFace - class CfgFaces.GravesHead_EP1.Face01 not found
 0:53:57 "TELEPORT REVERT for player UID#76561198103448555 from [12769.9,14832.9,0.00140381] to [5054.74,9759.63,0.00143433], 9235 meters, now at [12769.9,14832.9,0.00140381]"
 0:54:50 "TELEPORT REVERT for player UID#76561198103448555 from [12770,14833.1,0.0014801] to [7044.28,8679.09,0.338104], 8408 meters, now at [12770,14833.1,0.0014801]"

This all takes place without me EVER using the "Return player to last pos" option.

The log does not spam, it shows as above once only.

If I try to teleport again the player will NOT teleport.  He just appears at my location for a second that he's back at the original occasion again. (ie the location he was when he first logged in)
(Server) rpt says:

 1:04:13 "TELEPORT REVERT for player UID#76561198103448555 from [12770,14833.1,0.00141907] to [7042.52,8679.32,0.434357], 8409 meters, now at [12770,14833.1,0.00141907]"

 1:04:13 "TELEPORT REVERT for player UID#76561198103448555 from [12770,14833.1,0.00141907] to [7042.26,8679.46,0.0017395], 8409 meters, now at [12770,14833.1,0.00141907]"

**Comment by noxsicarius** [comment]
So these logs don't appear until the second attempt?

**Comment by 2000jago** [comment]
Correct.
However, I just restarted and did EXACTLY the same thing again.  This time the user did not teleport on the first attempt.  Log said:

 1:26:24 SetFace error: class CfgFaces.GravesHead_EP1.Face01 not found
 1:26:24 Error: Error during SetFace - class CfgFaces.GravesHead_EP1.Face01 not found

 1:26:53 "TELEPORT REVERT for player UID#76561198103448555 from [12769.9,14833.4,0.00138855] to [4162.33,10876.7,0.00146484], 9475 meters, now at [12769.9,14833.4,0.00138855]"

 1:26:53 "TELEPORT REVERT for player UID#76561198103448555 from [12769.9,14833.4,0.00138855] to [4162.09,10876.9,0.00143433], 9475 meters, now at [12769.9,14833.4,0.00138855]"

Tried again, same result, log (server side) said:

 1:31:23 "TELEPORT REVERT for player UID#76561198103448555 from [12769.9,14833.4,0.00141907] to [4161.13,10879.7,0.00143433], 9475 meters, now at [12769.9,14833.4,0.00141907]"

 1:31:23 "TELEPORT REVERT for player UID#76561198103448555 from [12769.9,14833.4,0.00141907] to [4161.13,10879.7,0.00143433], 9475 meters, now at [12769.9,14833.4,0.00141907]"

The revert option was never used since user never left the position he was upon login.

**Comment by noxsicarius** [comment]
Try upping the sleep a little bit on line 33 of the teleport file. Change it from 1 to 2. I am thinking there needs to be a little longer of a wait to allow the game to register the change.

**Comment by 2000jago** [comment]
Ok, TPtoME.sqf, changed 1 to 2 on line 33 but no change aside from taking a little longer for the user to flash in front of me before being returned to his original location.
Log: (first attempt)
 1:42:08 SetFace error: class CfgFaces.GravesHead_EP1.Face01 not found

 1:42:08 Error: Error during SetFace - class CfgFaces.GravesHead_EP1.Face01 not found

 1:42:30 "TELEPORT REVERT for player UID#76561198103448555 from [12769.9,14833.4,0.00146484] to [4155.93,10878.7,0.00143433], 9480 meters, now at [12769.9,14833.4,0.00146484]"

 1:42:30 "TELEPORT REVERT for player UID#76561198103448555 from [12769.9,14833.4,0.00146484] to [4157.18,10878.1,0.00143433], 9479 meters, now at [12769.9,14833.4,0.00146484]"

Log: (Second attempt)
1:42:50 "TELEPORT REVERT for player UID#76561198103448555 from [12769.9,14833.4,0.00138855] to [4156.47,10878.3,0.00143433], 9480 meters, now at [12769.9,14833.4,0.00138855]"

**Comment by noxsicarius** [comment]
I am looking into this. It's just an odd thing to have this happen randomly like that. Your init still uses the admintool antihack right?

**Comment by noxsicarius** [comment]
If so, lets see if this could have anything to do with the return teleport having some kind of unthrown error. To do this go back to the teleport file where you changed the sleep, delete line 26 (EAT_returnPlayer........). This seems like a new issue and I don't htink that line can cause this, but might as well give it a test to be sure.

**Comment by 2000jago** [comment]
Yes, still using admintool antihack.
Removing like 26 did not affect the issue.  It still does exactly the same thing with the line removed.
The only changes made to the server recently are the uninstalling of the R3F tow and lift and installation of the Nighmare version.  Perhaps I should do a complete reinstall of testbranch admintools ?

**Comment by noxsicarius** [comment]
I don't know if that would help. I don't see what is going on with this. The log shows the player position was manipulated from the antihack, however the bypass is the same in the main test branch. The only real difference is the event handler setup. I cleaned it up, but it is still the same code with one less step and that worked for you previously.

**Comment by 2000jago** [comment]
Is there anything else I could try that may reveal more clues as to the source of the problem?  I'd like to avoid having to go back to the master branch as I'm finding some of the new features of the testbranch very useful (the fast run in particular).  But TP to ME is something I use a lot so I kinda need to get it working.  Happy to try any other suggestions you may have.

**Comment by noxsicarius** [comment]
You could try a reinstall of it I guess. I really don't see what would cause this out of nowhere like that.

**Comment by 2000jago** [comment]
OK, going to do so now and will report back.

**Comment by 2000jago** [comment]
Okay re-uploaded all files from a fresh download off the github.  All files replaced except my custom crates and my config file.

Went through all instructions to make sure everything is where it should be and found this from step 18.
if (!_parachuteWest and !(locked _object)) then {

Searching for this string was not found because "&&" was used and not "and".

My compile/server_updateObject.sqf is slightly different:

BEFORE:

// Epoch Admin Tools
if (_object getVariable "MalSar" == 1) exitWith {};
if (!_parachuteWest && !(locked _object)) then {
	if (_objectID == "0" && _uid == "0") then
	{
		_object_position = getPosATL _object;
    	_isNotOk = true;
	};
};

I changed mine to be the same as yours (from directions on GitHub page)

AFTER:

// Epoch Admin Tools
if (_object getVariable ["MalSar",0] == 1) exitWith {};
if (!_parachuteWest and !(locked _object)) then {
	if (_objectID == "0" && _uid == "0") then
	{
		_object_position = getPosATL _object;
    	_isNotOk = true;
	};
};
==================


However, even after clean install, the problem persists.  I get:
14:33:53 "TELEPORT REVERT for player UID#76561198103448555 from [12733.8,14805.5,0.00120544] to [12694.4,14804.6,0.000656128], 39 meters, now at [12733.8,14805.5,0.00120544]"

14:33:54 "TELEPORT REVERT for player UID#76561198103448555 from [12733.8,14805.5,0.00120544] to [12694.4,14804.6,0.0014801], 39 meters, now at [12733.8,14805.5,0.00120544]"

in the server rpt as soon as I try teleport player to me.  He appears briefly then is back at his original location.

Here's what the client rpt is saying in case that offers any clues pertaining to this issue:

"DEBUG: loadscreen guard started."
"Admin Tools: config.sqf loaded"
"Admin Tools: variables.sqf loaded"
"Res3tting B!S effects..."
"\z\addons\dayz_code\system\REsec.sqf:Monitoring Remote Exe..."
Cannot use magazine 30Rnd_556x45_G36 in muzzle Binocular
Warning: z\addons\dayz_communityassets\models\compass.p3d:0 Error while trying to generate ST for points: 863, 853, 852
No owner
No owner
No owner
"PLAYER RESULT: true"
No owner
No owner
No owner
No owner
No owner
Cannot use magazine PG7V in muzzle M14_EP1
SetFace error: class CfgFaces.GravesHead_EP1.Face01 not found
Error: Error during SetFace - class CfgFaces.GravesHead_EP1.Face01 not found
"PLOGIN: Login loop completed!"
"Admin Tools: Activate.sqf loaded"
Warning: looped for animation: ca\anims\characters\data\anim\sdr\mov\erc\stp\non\non\amovpercmstpsnonwnondnon_amovpercmstpsraswpstdnon_end.rtm differs (looped now 0)! MoveName: amovpercmstpsnonwnondnon_amovpercmstpsraswpstdnon_end
"Admin Tools: safeZones.sqf loaded"
"Admin Tools: NumberKeys.sqf Loaded"
"Admin Tools: FunctionKeys.sqf Loaded"
Warning: z\addons\dayz_communityassets\models\razor.p3d:0 Error while trying to generate ST for points: 214, 349, 208

**Comment by noxsicarius** [comment]
Going to be out all day, I will try to fix this soon.

**Comment by 2000jago** [comment]
Update:  Not sure if this information will be of any use, but whilst playing today I noticed the "teleport revert for player...etc" message appear in the logs.  Exactly as it would if I had attempted to TP a player.  However, I had not used ANY admin tools at the time it showed up.
Could perhaps have been a player trying some kind of teleport hack that was blocked by the antihack?  It only appeared once in the logs and hasn't occurred again since.  Just thought I'd mention it.

**Comment by noxsicarius** [comment]
I'll explain that in a sec. I might have found the issue.

**Comment by noxsicarius** [comment]
Not the issue. Do me a favor. Try to teleport them using the support ticket system. They need to have the action menu active, then they need to request help. You can then open the ticket menu to teleport to you. This will tell me if it is the teleport file or the bypass I hope.

**Comment by noxsicarius** [comment]
As for your post: it could be a teleport or the player had high latency and it did not send the server it's new position so it saw the player move 3 ticks since the client didn't send two ticks of data.

**Comment by 2000jago** [comment]
I'm on it.  Will report back shortly.

**Comment by 2000jago** [comment]
Tested:  I had to enable it first as I don't normally have it enabled.  Once enabled however, user creates a ticket.  I get the notification, go to ticket menu say "TP to me" select users name and he appears for a second then back to his location.  Log says:
 0:47:59 "TELEPORT REVERT for player UID#76561198103448555 from [11722.7,14830.2,0.00135803] to [6556.97,14085.6,-0.0493164], 5222 meters, now at [11722.7,14830.2,0.00135803]"

 0:48:00 "TELEPORT REVERT for player UID#76561198103448555 from [11722.7,14830.2,0.00135803] to [6556.97,14085.6,0.00137329], 5222 meters, now at [11722.7,14830.2,0.00135803]"

 0:48:43 "TELEPORT REVERT for player UID#76561198103448555 from [11722.7,14830.2,0.00149536] to [6557.73,14084.3,-0.0518494], 5222 meters, now at [11722.7,14830.2,0.00149536]"

That is the result of TWO attempts.  The first two entries appear after the first try and the 3rd entry after the second try.

**Comment by noxsicarius** [comment]
Try changing that sleep line in the normal teleport to 5. If that does not work we know it isn't a lag issue and that the problem most likely comes from the variable setting and pushing to client.

**Comment by 2000jago** [comment]
Ok, changed to 5.  Same result. I tried normal TP to me as well as through ticket system.
Same result.

**Comment by noxsicarius** [comment]
Alright. Looks like it's an issue with getting the variable propagated.

**Comment by noxsicarius** [comment]
Replace your variables with this and we will see if that fixes the issue. http://pastebin.com/zv5imdMq

**Comment by 2000jago** [comment]
That works! 
Works via normal TP and via ticket system. 
Return to last location works for normal TP, but not from ticket system though.

**Comment by noxsicarius** [comment]
It's not set up for the ticket system yet. It was still in testing and I didn't want to have to worry about troubleshooting two files.

**Comment by 2000jago** [comment]
No worries, at least you're on the correct trail of this issue now.  Let me know if you need anything else tested.  It's 1:30am for me right now, but I'll still be up a while.  ;)

**Comment by noxsicarius** [comment]
Do me a favor and test the server message when you get a chance. Just choose the send server message from the admin menu. Type whatever you want in the box that shows up and just ask the players if they see the message on their screen. It uses a similar event handler that the teleport used. Since the teleport had issues I want to be sure the message works.

**Comment by noxsicarius** [comment]
Oh, you are 12 hours ahead of me then. This issue above isn't pressing. I don't need data until late sunday or even monday. Just whenever you have the time.

**Comment by 2000jago** [comment]
No problem.  Tested and confirmed working just fine.  Users see the message.

I am in Thailand btw. My server is hosted n Singapore.

**Comment by noxsicarius** [comment]
That is really crazy. It uses the same feature as the teleport one did, but for some reason the teleport one didn't work. It seems as if a server event handler can only send commands so you have to use a client handler to send variables.

**Comment by 2000jago** [comment]
Been stress testing today and can confirm it works reliably.  

Unrelated:
Is there any way to add "View Distance" options to the Actions Menu?  The reason I ask is that I have to have WPD's action menu installed to achieve this functionality.  Pretty much ALL the other WPD function's are already present in the admin tools options.
I don't want to have TWO action menu's so end up having to disable the Admin tools one, loosing the "fun" options (dance etc).  JUST so I can get the view distance ability for users who ask for it constantly.

I know this isn't the right place for "requests", so I apologize in advance.  Please direct me to a more appropriate way to submit one and I will do so.  ;)

**Comment by noxsicarius** [comment]
https://github.com/noxsicarius/Epoch-Admin-Tools/issues/296
